hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -796,6 +819,19 @@ public long countByRsqlAndCompatible(final String targetFilterQuery, final Long
         return JpaManagementHelper.countBySpec(targetRepository, specList);
     }
 
+    @Override
+    public long countByFailedInRolloutAndCompatible(final String rolloutId, final Long dsTypeId, final Long createdAt) {
+        final List<Specification<JpaTarget>> specList = Arrays.asList(
+            TargetSpecifications.failedActionsForRollout(rolloutId),
+            TargetSpecifications.isCompatibleWithDistributionSetType(dsTypeId));
+
+        if (createdAt != null) {
+            specList.add(TargetSpecifications.createdBefore(createdAt));
```

Comment:
why createdBefore is used - rollout fail filter would not contain devices created after anyway